python/taichi/aot/utils.py

```diff
@@ -62,12 +64,14 @@ def produce_injected_args(kernel, symbolic_args=None):
                 raise TaichiCompilationError(
                     f'{field_dim} from Arg {arg.name} doesn\'t match kernel\'s annotated field_dim={anno.field_dim}'
                 )
-
-            if anno.dtype is not None and not check_type_match(
-                    dtype, anno.dtype):
-                raise TaichiCompilationError(
-                    f' Arg {arg.name}\'s dtype {dtype.to_string()} doesn\'t match kernel\'s annotated dtype={anno.dtype.to_string()}'
-                )
+            anno_dtype = anno.dtype
+            if isinstance(anno.dtype, MatrixType):
```

Comment:
nit: we need to be consistent of using `anno_dtype` or `anno.dtype`, it might cause confusions.

Reply:
Thanks!